.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,27 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/ubuntu
+{
+	"name": "Ubuntu",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "mcr.microsoft.com/devcontainers/base:jammy",
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	"features": {
+		"ghcr.io/devcontainers/features/rust:1": {
+			"version": "latest",
+			"profile": "default",
+			"targets": "aarch64-apple-darwin,aarch64-unknown-linux-musl,x86_64-pc-windows-msvc,x86_64-unknown-linux-gnu,x86_64-unknown-linux-musl,x86_64-unknown-none"
+		}
+	},
+
+	// Allow kvm by setting privileged to true.
+	"privileged": true,
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "cargo xflowey restore-packages"
```

Comment:
never used devcontainers, so apologies if this is a silly question, but how "efficient" are things like "postCreateCommand", "features", etc...

are they run by each user using the dev container locally, or is there some docker-style layer-level caching of the results of these operations?

Reply:
See their docs, but `PostCreateCommand` is run once after the container is created. Generally it seems like these are repo-specific setup commands.

I think if we had our own custom docker file, there would be more caching, but tbh it doesn't take too long to setup anyways. There's not much we need to install these days.

Reply:
gotcha, makes sense.